sdks/python/apache_beam/io/textio.py

```diff
@@ -781,6 +782,13 @@ def __init__(
     """
 
     super().__init__(**kwargs)
+    if file_pattern:
+      try:
+        if not os.path.dirname(file_pattern):
+          file_pattern = os.path.join('.', file_pattern)
+      except TypeError:
```

Comment:
Seems like a very broad exception catch. What is the failure mode that you are handling?

Reply:
file_pattern coud be passed in as other types, for example, ValueProvider.